app/main.py

```diff
@@ -1,34 +1,126 @@
+from dataclasses import dataclass, field
+
+
+@dataclass
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
 
+    row: int
+    column: int
+    is_alive: bool = True
 
+
+@dataclass
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    start: tuple
+    end: tuple
+    is_drowned: bool = False
+    decks: list = field(init=False)
+
+    def __post_init__(self) -> None:
+        decks = []
+        for x_coord in range(self.start[0], self.end[0] + 1):
+            for y_coord in range(self.start[1], self.end[1] + 1):
+                decks.append(Deck(x_coord, y_coord))
+        self.decks = decks
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
```

Comment:
There should be a check to ensure that `deck` is not `None` before attempting to set `deck.is_alive = False`. If `get_deck` returns `None`, this will raise an error.